history.py

```diff
@@ -0,0 +1,30 @@
+from catalog import models as cmod
+
+class LastFiveMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+        # One-time configuration and initialization.
+
+    def __call__(self, request):
+        # Code to be executed for each request before
+        # the view (and later middleware) are called.
+        product_ids = request.session.get('last_five', [])
+        request.last_five = []
+        for p in product_ids:
+            request.last_five.append(cmod.Product.objects.get(id=   p))
+
+
+        response = self.get_response(request)
+
+        # Code to be executed for each request/response after
+        # the view is called.
+        product_ids = []
+        for p in request.last_five:
+            product_ids.append(p.id)
+
+        if len(product_ids) > 6:
```

Comment:
the issue here might be if len == 8, it will only pop one item.... that's why i have if len(prod_ids) = 7. But idk how you would get it to len == 8.... but the user might find a way.

Reply:
maybe a do while would be better then. 
while len(product_ids) > 6: